From 0824c11d5bb49da28ba3b9fd915ff4eb720fcb2d Mon Sep 17 00:00:00 2001
From: aiaimimi0920 <153103332+aiaimimi0920@users.noreply.github.com>
Date: Sun, 30 Aug 2026 20:03:59 +0800
Subject: [PATCH] build: add Fortune Wheel profiles

Carry the current Web and Windows feature-reduction profiles using the modern .gdbuild extension; the deleted legacy lucky_wheel profile is intentionally not restored.

Godot-Patch-Topic: build.profiles.fortune-wheel
---
 fortune_wheel_web.gdbuild     | 342 ++++++++++++++++++++++++++++++++++
 fortune_wheel_windows.gdbuild | 341 +++++++++++++++++++++++++++++++++
 2 files changed, 683 insertions(+)
 create mode 100644 fortune_wheel_web.gdbuild
 create mode 100644 fortune_wheel_windows.gdbuild

diff --git a/fortune_wheel_web.gdbuild b/fortune_wheel_web.gdbuild
new file mode 100644
index 0000000000000000000000000000000000000000..96236a5cc0d4ed1496cc3c926762d89878d39266
--- /dev/null
+++ b/fortune_wheel_web.gdbuild
@@ -0,0 +1,342 @@
+{
+	"disabled_build_options": {
+		"d3d12": false,
+		"disable_3d": true,
+		"disable_navigation_2d": true,
+		"disable_navigation_3d": true,
+		"disable_physics_2d": true,
+		"disable_physics_3d": true,
+		"disable_xr": true,
+		"forward_mobile_renderer": false,
+		"metal": false,
+		"module_godot_physics_2d_enabled": false,
+		"module_godot_physics_3d_enabled": false,
+		"module_jolt_physics_enabled": false,
+		"module_openxr_enabled": false,
+		"vulkan": false,
+		"wayland": false,
+		"x11": false
+	},
+	"disabled_classes": [
+		"AESContext",
+		"AStar2D",
+		"AStar3D",
+		"AStarGrid2D",
+		"AnimatedTexture",
+		"AnimationNodeBlendSpace1D",
+		"AnimationNodeBlendSpace2D",
+		"AnimationNodeExtension",
+		"AnimationNodeOutput",
+		"AnimationNodeStateMachinePlayback",
+		"AnimationNodeSync",
+		"AnimationNodeTimeScale",
+		"AtlasTexture",
+		"AudioEffectAmplify",
+		"AudioEffectBandLimitFilter",
+		"AudioEffectBandPassFilter",
+		"AudioEffectCapture",
+		"AudioEffectChorus",
+		"AudioEffectCompressor",
+		"AudioEffectDelay",
+		"AudioEffectDistortion",
+		"AudioEffectEQ",
+		"AudioEffectHardLimiter",
+		"AudioEffectHighPassFilter",
+		"AudioEffectHighShelfFilter",
+		"AudioEffectInstance",
+		"AudioEffectLimiter",
+		"AudioEffectLowShelfFilter",
+		"AudioEffectNotchFilter",
+		"AudioEffectPanner",
+		"AudioEffectPhaser",
+		"AudioEffectPitchShift",
+		"AudioEffectRecord",
+		"AudioEffectStereoEnhance",
+		"AudioListener2D",
+		"AudioSample",
+		"AudioSamplePlayback",
+		"AudioStreamGenerator",
+		"AudioStreamInteractive",
+		"AudioStreamMicrophone",
+		"AudioStreamPlayback",
+		"AudioStreamPlaylist",
+		"AudioStreamPolyphonic",
+		"AudioStreamRandomizer",
+		"AudioStreamSynchronized",
+		"BackBufferCopy",
+		"BaseMaterial3D",
+		"Bone2D",
+		"BoneMap",
+		"CPUParticles2D",
+		"Camera2D",
+		"CameraAttributes",
+		"CameraFeed",
+		"CameraServer",
+		"CameraTexture",
+		"CanvasGroup",
+		"CanvasTexture",
+		"ClassDB",
+		"CodeHighlighter",
+		"CollisionObject2D",
+		"ColorPalette",
+		"ColorPicker",
+		"Compositor",
+		"CompositorEffect",
+		"CryptoKey",
+		"Curve2D",
+		"Curve3D",
+		"CurveTexture",
+		"CurveXYZTexture",
+		"DPITexture",
+		"DTLSServer",
+		"ENetConnection",
+		"EncodedObjectAsID",
+		"EngineDebugger",
+		"EngineProfiler",
+		"ExternalTexture",
+		"FogMaterial",
+		"FoldableContainer",
+		"FoldableGroup",
+		"FramebufferCacheRD",
+		"GDExtension",
+		"GDExtensionManager",
+		"GDScriptEditorTranslationParserPlugin",
+		"GDScriptHelper",
+		"GDScriptNativeClass",
+		"GLTFAccessor",
+		"GLTFAnimation",
+		"GLTFBufferView",
+		"GLTFCamera",
+		"GLTFDocument",
+		"GLTFDocumentExtension",
+		"GLTFLight",
+		"GLTFMesh",
+		"GLTFNode",
+		"GLTFObjectModelProperty",
+		"GLTFPhysicsBody",
+		"GLTFPhysicsShape",
+		"GLTFSkeleton",
+		"GLTFSkin",
+		"GLTFSpecGloss",
+		"GLTFState",
+		"GLTFTexture",
+		"GLTFTextureSampler",
+		"GPUParticles2D",
+		"Geometry3D",
+		"GifExporter",
+		"GradientTexture2D",
+		"GraphEdit",
+		"GraphElement",
+		"GridContainer",
+		"IPWindows",
+		"ImageFormatLoader",
+		"ImporterMesh",
+		"InstancePlaceholder",
+		"ItemList",
+		"JNISingleton",
+		"JSONRPC",
+		"JavaClass",
+		"JavaClassWrapper",
+		"JavaObject",
+		"JavaScriptBridge",
+		"JavaScriptObject",
+		"Joint2D",
+		"KinematicCollision2D",
+		"KinematicCollision3D",
+		"LabelSettings",
+		"Light2D",
+		"LightOccluder2D",
+		"LightmapGIData",
+		"Lightmapper",
+		"Logger",
+		"Marker2D",
+		"Marshalls",
+		"Mesh",
+		"MeshConvexDecompositionSettings",
+		"MeshDataTool",
+		"MeshInstance2D",
+		"MeshLibrary",
+		"MeshTexture",
+		"MissingNode",
+		"MissingResource",
+		"MovieWriter",
+		"MultiMesh",
+		"MultiMeshInstance2D",
+		"MultiplayerAPI",
+		"MultiplayerSpawner",
+		"MultiplayerSynchronizer",
+		"Mutex",
+		"NativeMenu",
+		"NavigationAgent2D",
+		"NavigationAgent3D",
+		"NavigationLink2D",
+		"NavigationMesh",
+		"NavigationMeshGenerator",
+		"NavigationMeshSourceGeometryData2D",
+		"NavigationMeshSourceGeometryData3D",
+		"NavigationObstacle2D",
+		"NavigationPathQueryParameters2D",
+		"NavigationPathQueryParameters3D",
+		"NavigationPathQueryResult2D",
+		"NavigationPathQueryResult3D",
+		"NavigationPolygon",
+		"NavigationRegion2D",
+		"NavigationServer2D",
+		"NavigationServer3D",
+		"Node3D",
+		"Node3DGizmo",
+		"Occluder3D",
+		"OccluderPolygon2D",
+		"OggPacketSequencePlayback",
+		"OpenXRAPIExtension",
+		"OpenXRAction",
+		"OpenXRActionMap",
+		"OpenXRActionSet",
+		"OpenXRBindingModifier",
+		"OpenXRExtensionWrapper",
+		"OpenXRFutureResult",
+		"OpenXRHapticBase",
+		"OpenXRIPBinding",
+		"OpenXRInteractionProfile",
+		"OpenXRInteractionProfileMetadata",
+		"PCKPacker",
+		"PackedDataContainer",
+		"PackedDataContainerRef",
+		"PacketPeer",
+		"PanoramaSkyMaterial",
+		"Parallax2D",
+		"ParallaxBackground",
+		"ParallaxLayer",
+		"ParticleProcessMaterial",
+		"Path2D",
+		"PathFollow2D",
+		"Performance",
+		"PhysicalSkyMaterial",
+		"PhysicsDirectBodyState2D",
+		"PhysicsDirectBodyState3D",
+		"PhysicsDirectSpaceState2D",
+		"PhysicsDirectSpaceState3D",
+		"PhysicsMaterial",
+		"PhysicsPointQueryParameters2D",
+		"PhysicsPointQueryParameters3D",
+		"PhysicsRayQueryParameters2D",
+		"PhysicsRayQueryParameters3D",
+		"PhysicsServer2D",
+		"PhysicsServer2DManager",
+		"PhysicsServer3D",
+		"PhysicsServer3DManager",
+		"PhysicsServer3DRenderingServerHandler",
+		"PhysicsShapeQueryParameters2D",
+		"PhysicsShapeQueryParameters3D",
+		"PhysicsTestMotionParameters2D",
+		"PhysicsTestMotionParameters3D",
+		"PhysicsTestMotionResult2D",
+		"PhysicsTestMotionResult3D",
+		"PlaceholderMaterial",
+		"PlaceholderTexture2D",
+		"PolygonPathFinder",
+		"PortableCompressedTexture2D",
+		"ProceduralSkyMaterial",
+		"Process",
+		"RDAttachmentFormat",
+		"RDFramebufferPass",
+		"RDPipelineColorBlendState",
+		"RDPipelineColorBlendStateAttachment",
+		"RDPipelineDepthStencilState",
+		"RDPipelineMultisampleState",
+		"RDPipelineRasterizationState",
+		"RDPipelineSpecializationConstant",
+		"RDSamplerState",
+		"RDShaderFile",
+		"RDShaderSPIRV",
+		"RDShaderSource",
+		"RDTextureFormat",
+		"RDTextureView",
+		"RDUniform",
+		"RDVertexAttribute",
+		"RayCast2D",
+		"RemoteTransform2D",
+		"RenderData",
+		"RenderSceneBuffers",
+		"RenderSceneBuffersConfiguration",
+		"RenderSceneData",
+		"RenderingDevice",
+		"RenderingServer",
+		"ResourceFormatLoader",
+		"ResourceFormatSaver",
+		"ResourcePreloader",
+		"SceneReplicationConfig",
+		"SceneState",
+		"ScriptBacktrace",
+		"ScriptExtension",
+		"ScriptLanguage",
+		"Semaphore",
+		"ShaderGlobalsOverride",
+		"ShaderInclude",
+		"ShaderIncludeDB",
+		"Shape2D",
+		"Shape3D",
+		"ShapeCast2D",
+		"Skeleton2D",
+		"SkeletonModification2D",
+		"SkeletonModificationStack2D",
+		"SkeletonProfile",
+		"Skin",
+		"SkinReference",
+		"Slider",
+		"SpinBox",
+		"SplitContainer",
+		"StatusIndicator",
+		"Steam",
+		"StreamPeer",
+		"SubtweenTweener",
+		"SurfaceTool",
+		"SystemFont",
+		"TCPServer",
+		"TLSOptions",
+		"TabContainer",
+		"TextLine",
+		"TextParagraph",
+		"TextServerExtension",
+		"Texture2DRD",
+		"Texture3D",
+		"TextureLayered",
+		"ThemeContext",
+		"Thread",
+		"TileData",
+		"TileMap",
+		"TileMapLayer",
+		"TileMapPattern",
+		"TileSet",
+		"TileSetSource",
+		"TouchScreenButton",
+		"TranslationDomain",
+		"TreeItem",
+		"TriangleMesh",
+		"UDPServer",
+		"UPNP",
+		"UPNPDevice",
+		"UndoRedo",
+		"UniformSetCacheRD",
+		"VideoStreamPlayback",
+		"VideoStreamTheora",
+		"VisibleOnScreenNotifier2D",
+		"VisualShader",
+		"VisualShaderNode",
+		"VoxelGIData",
+		"WebRTCPeerConnection",
+		"WorkerThreadPool",
+		"World2D",
+		"World3D",
+		"WorldEnvironment",
+		"X509Certificate",
+		"XMLParser",
+		"XRInterface",
+		"XRPose",
+		"XRServer",
+		"XRTracker",
+		"XRVRS",
+		"ZIPPacker"
+	],
+	"type": "build_profile"
+}
\ No newline at end of file
diff --git a/fortune_wheel_windows.gdbuild b/fortune_wheel_windows.gdbuild
new file mode 100644
index 0000000000000000000000000000000000000000..2b5d58abcf2c45b9e10e06afd3cbfbf4e83c2d1e
--- /dev/null
+++ b/fortune_wheel_windows.gdbuild
@@ -0,0 +1,341 @@
+{
+	"disabled_build_options": {
+		"disable_3d": true,
+		"disable_navigation_2d": true,
+		"disable_navigation_3d": true,
+		"disable_physics_2d": true,
+		"disable_physics_3d": true,
+		"disable_xr": true,
+		"forward_mobile_renderer": false,
+		"metal": false,
+		"module_godot_physics_2d_enabled": false,
+		"module_godot_physics_3d_enabled": false,
+		"module_jolt_physics_enabled": false,
+		"module_openxr_enabled": false,
+		"opengl3": false,
+		"wayland": false,
+		"x11": false
+	},
+	"disabled_classes": [
+		"AESContext",
+		"AStar2D",
+		"AStar3D",
+		"AStarGrid2D",
+		"AnimatedTexture",
+		"AnimationNodeBlendSpace1D",
+		"AnimationNodeBlendSpace2D",
+		"AnimationNodeExtension",
+		"AnimationNodeOutput",
+		"AnimationNodeStateMachinePlayback",
+		"AnimationNodeSync",
+		"AnimationNodeTimeScale",
+		"AtlasTexture",
+		"AudioEffectAmplify",
+		"AudioEffectBandLimitFilter",
+		"AudioEffectBandPassFilter",
+		"AudioEffectCapture",
+		"AudioEffectChorus",
+		"AudioEffectCompressor",
+		"AudioEffectDelay",
+		"AudioEffectDistortion",
+		"AudioEffectEQ",
+		"AudioEffectHardLimiter",
+		"AudioEffectHighPassFilter",
+		"AudioEffectHighShelfFilter",
+		"AudioEffectInstance",
+		"AudioEffectLimiter",
+		"AudioEffectLowShelfFilter",
+		"AudioEffectNotchFilter",
+		"AudioEffectPanner",
+		"AudioEffectPhaser",
+		"AudioEffectPitchShift",
+		"AudioEffectRecord",
+		"AudioEffectStereoEnhance",
+		"AudioListener2D",
+		"AudioSample",
+		"AudioSamplePlayback",
+		"AudioStreamGenerator",
+		"AudioStreamInteractive",
+		"AudioStreamMicrophone",
+		"AudioStreamPlayback",
+		"AudioStreamPlaylist",
+		"AudioStreamPolyphonic",
+		"AudioStreamRandomizer",
+		"AudioStreamSynchronized",
+		"BackBufferCopy",
+		"BaseMaterial3D",
+		"Bone2D",
+		"BoneMap",
+		"CPUParticles2D",
+		"Camera2D",
+		"CameraAttributes",
+		"CameraFeed",
+		"CameraServer",
+		"CameraTexture",
+		"CanvasGroup",
+		"CanvasTexture",
+		"ClassDB",
+		"CodeHighlighter",
+		"CollisionObject2D",
+		"ColorPalette",
+		"ColorPicker",
+		"Compositor",
+		"CompositorEffect",
+		"CryptoKey",
+		"Curve2D",
+		"Curve3D",
+		"CurveTexture",
+		"CurveXYZTexture",
+		"DPITexture",
+		"DTLSServer",
+		"ENetConnection",
+		"EncodedObjectAsID",
+		"EngineDebugger",
+		"EngineProfiler",
+		"ExternalTexture",
+		"FogMaterial",
+		"FoldableContainer",
+		"FoldableGroup",
+		"FramebufferCacheRD",
+		"GDExtension",
+		"GDExtensionManager",
+		"GDScriptEditorTranslationParserPlugin",
+		"GDScriptHelper",
+		"GDScriptNativeClass",
+		"GLTFAccessor",
+		"GLTFAnimation",
+		"GLTFBufferView",
+		"GLTFCamera",
+		"GLTFDocument",
+		"GLTFDocumentExtension",
+		"GLTFLight",
+		"GLTFMesh",
+		"GLTFNode",
+		"GLTFObjectModelProperty",
+		"GLTFPhysicsBody",
+		"GLTFPhysicsShape",
+		"GLTFSkeleton",
+		"GLTFSkin",
+		"GLTFSpecGloss",
+		"GLTFState",
+		"GLTFTexture",
+		"GLTFTextureSampler",
+		"GPUParticles2D",
+		"Geometry3D",
+		"GifExporter",
+		"GradientTexture2D",
+		"GraphEdit",
+		"GraphElement",
+		"GridContainer",
+		"IPWindows",
+		"ImageFormatLoader",
+		"ImporterMesh",
+		"InstancePlaceholder",
+		"ItemList",
+		"JNISingleton",
+		"JSONRPC",
+		"JavaClass",
+		"JavaClassWrapper",
+		"JavaObject",
+		"JavaScriptBridge",
+		"JavaScriptObject",
+		"Joint2D",
+		"KinematicCollision2D",
+		"KinematicCollision3D",
+		"LabelSettings",
+		"Light2D",
+		"LightOccluder2D",
+		"LightmapGIData",
+		"Lightmapper",
+		"Logger",
+		"Marker2D",
+		"Marshalls",
+		"Mesh",
+		"MeshConvexDecompositionSettings",
+		"MeshDataTool",
+		"MeshInstance2D",
+		"MeshLibrary",
+		"MeshTexture",
+		"MissingNode",
+		"MissingResource",
+		"MovieWriter",
+		"MultiMesh",
+		"MultiMeshInstance2D",
+		"MultiplayerAPI",
+		"MultiplayerSpawner",
+		"MultiplayerSynchronizer",
+		"Mutex",
+		"NativeMenu",
+		"NavigationAgent2D",
+		"NavigationAgent3D",
+		"NavigationLink2D",
+		"NavigationMesh",
+		"NavigationMeshGenerator",
+		"NavigationMeshSourceGeometryData2D",
+		"NavigationMeshSourceGeometryData3D",
+		"NavigationObstacle2D",
+		"NavigationPathQueryParameters2D",
+		"NavigationPathQueryParameters3D",
+		"NavigationPathQueryResult2D",
+		"NavigationPathQueryResult3D",
+		"NavigationPolygon",
+		"NavigationRegion2D",
+		"NavigationServer2D",
+		"NavigationServer3D",
+		"Node3D",
+		"Node3DGizmo",
+		"Occluder3D",
+		"OccluderPolygon2D",
+		"OggPacketSequencePlayback",
+		"OpenXRAPIExtension",
+		"OpenXRAction",
+		"OpenXRActionMap",
+		"OpenXRActionSet",
+		"OpenXRBindingModifier",
+		"OpenXRExtensionWrapper",
+		"OpenXRFutureResult",
+		"OpenXRHapticBase",
+		"OpenXRIPBinding",
+		"OpenXRInteractionProfile",
+		"OpenXRInteractionProfileMetadata",
+		"PCKPacker",
+		"PackedDataContainer",
+		"PackedDataContainerRef",
+		"PacketPeer",
+		"PanoramaSkyMaterial",
+		"Parallax2D",
+		"ParallaxBackground",
+		"ParallaxLayer",
+		"ParticleProcessMaterial",
+		"Path2D",
+		"PathFollow2D",
+		"Performance",
+		"PhysicalSkyMaterial",
+		"PhysicsDirectBodyState2D",
+		"PhysicsDirectBodyState3D",
+		"PhysicsDirectSpaceState2D",
+		"PhysicsDirectSpaceState3D",
+		"PhysicsMaterial",
+		"PhysicsPointQueryParameters2D",
+		"PhysicsPointQueryParameters3D",
+		"PhysicsRayQueryParameters2D",
+		"PhysicsRayQueryParameters3D",
+		"PhysicsServer2D",
+		"PhysicsServer2DManager",
+		"PhysicsServer3D",
+		"PhysicsServer3DManager",
+		"PhysicsServer3DRenderingServerHandler",
+		"PhysicsShapeQueryParameters2D",
+		"PhysicsShapeQueryParameters3D",
+		"PhysicsTestMotionParameters2D",
+		"PhysicsTestMotionParameters3D",
+		"PhysicsTestMotionResult2D",
+		"PhysicsTestMotionResult3D",
+		"PlaceholderMaterial",
+		"PlaceholderTexture2D",
+		"PolygonPathFinder",
+		"PortableCompressedTexture2D",
+		"ProceduralSkyMaterial",
+		"Process",
+		"RDAttachmentFormat",
+		"RDFramebufferPass",
+		"RDPipelineColorBlendState",
+		"RDPipelineColorBlendStateAttachment",
+		"RDPipelineDepthStencilState",
+		"RDPipelineMultisampleState",
+		"RDPipelineRasterizationState",
+		"RDPipelineSpecializationConstant",
+		"RDSamplerState",
+		"RDShaderFile",
+		"RDShaderSPIRV",
+		"RDShaderSource",
+		"RDTextureFormat",
+		"RDTextureView",
+		"RDUniform",
+		"RDVertexAttribute",
+		"RayCast2D",
+		"RemoteTransform2D",
+		"RenderData",
+		"RenderSceneBuffers",
+		"RenderSceneBuffersConfiguration",
+		"RenderSceneData",
+		"RenderingDevice",
+		"RenderingServer",
+		"ResourceFormatLoader",
+		"ResourceFormatSaver",
+		"ResourcePreloader",
+		"SceneReplicationConfig",
+		"SceneState",
+		"ScriptBacktrace",
+		"ScriptExtension",
+		"ScriptLanguage",
+		"Semaphore",
+		"ShaderGlobalsOverride",
+		"ShaderInclude",
+		"ShaderIncludeDB",
+		"Shape2D",
+		"Shape3D",
+		"ShapeCast2D",
+		"Skeleton2D",
+		"SkeletonModification2D",
+		"SkeletonModificationStack2D",
+		"SkeletonProfile",
+		"Skin",
+		"SkinReference",
+		"Slider",
+		"SpinBox",
+		"SplitContainer",
+		"StatusIndicator",
+		"Steam",
+		"StreamPeer",
+		"SubtweenTweener",
+		"SurfaceTool",
+		"SystemFont",
+		"TCPServer",
+		"TLSOptions",
+		"TabContainer",
+		"TextLine",
+		"TextParagraph",
+		"TextServerExtension",
+		"Texture2DRD",
+		"Texture3D",
+		"TextureLayered",
+		"ThemeContext",
+		"Thread",
+		"TileData",
+		"TileMap",
+		"TileMapLayer",
+		"TileMapPattern",
+		"TileSet",
+		"TileSetSource",
+		"TouchScreenButton",
+		"TranslationDomain",
+		"TreeItem",
+		"TriangleMesh",
+		"UDPServer",
+		"UPNP",
+		"UPNPDevice",
+		"UndoRedo",
+		"UniformSetCacheRD",
+		"VideoStreamPlayback",
+		"VideoStreamTheora",
+		"VisibleOnScreenNotifier2D",
+		"VisualShader",
+		"VisualShaderNode",
+		"VoxelGIData",
+		"WebRTCPeerConnection",
+		"WorkerThreadPool",
+		"World2D",
+		"World3D",
+		"WorldEnvironment",
+		"X509Certificate",
+		"XMLParser",
+		"XRInterface",
+		"XRPose",
+		"XRServer",
+		"XRTracker",
+		"XRVRS",
+		"ZIPPacker"
+	],
+	"type": "build_profile"
+}
\ No newline at end of file

base-commit: 5ec4857b340b6284a18b49b2eda462bd250f219a
prerequisite-patch-id: b38459b409a3326aed65d2433e1341638171f884
prerequisite-patch-id: bdb0fe8d6ba7fff1b900ae33d9bec552ea61724c
prerequisite-patch-id: 4dcdddc73066f768cb97c56444abacb21bda061c
prerequisite-patch-id: 1f2fbc55f35da3e943dcbaeeb3c9a6d798621bf3
prerequisite-patch-id: 4dd1cec6888d5160c85be2072b9e5dd05411d007
prerequisite-patch-id: 60110ae09a2d09796ad65e339ee0b5abc2ce541a
prerequisite-patch-id: bdb0025d4207a7cf0e1466ca96b2824d238b3caf
prerequisite-patch-id: 8c55a8201fec417bc2629b903ea2d630aedab1d8
prerequisite-patch-id: 9dfaaac9d4a47edf9d1eccce8fa66b69e674a911
prerequisite-patch-id: 17fa223ec74d20c7792e45826a4f0d8ecd7f85d7
prerequisite-patch-id: 0fb9ef780b3f10a7e82968e44defda508734fa91
prerequisite-patch-id: 2af824bf187ff13746bd620b1032f57d166c0114
